testsuite/tests/lib-printf/tprintf.ml

```diff
@@ -593,9 +604,16 @@ try
   test (sprintf "%*LX" 5 42L = "   2A");
   (*test (sprintf "%-0+ #*LX" 5 42L = "0X2A ");*)
     (* >> '-' is incompatible with '0' *)
+  test_roundtrip "0x%LX" Int64.of_string "0x0";
+  test_roundtrip "0x%LX" Int64.of_string "0x123";
+  test_roundtrip "0x%LX" Int64.of_string "0xABCDEF";
+  test_roundtrip "0x%LX" Int64.of_string "0x1234567812345678";
+  test_roundtrip "0x%LX" Int64.of_string "0x7FFFFFFFFFFFFFFF";
 
-  printf "\nLx negative\n%!";
+  printf "\nLX negative\n%!";
   test (sprintf "%LX" (-42L) = "FFFFFFFFFFFFFFD6");
+  test_roundtrip "0x%LX" Int64.of_string "0x8000000000000000";
+  test_roundtrip "0x%LX" Int64.of_string "0xFFFFFFFFFFFFFFFF";
```

Comment:
The tests for `Nativeint` seem to be missing. However, they could be added later: I am quite keen to have a version of 4.14 with all new deprecation warnings.

Reply:
Yes, there is no existing tests for `Nativeint`, as they are harder to write (the tests depend on whether you are on 32- or 64-bits). I wasn't planning to add them...

https://github.com/ocaml/ocaml/blob/3aaa9a7cf52b6980a850bfafde724157a783b98d/testsuite/tests/lib-printf/tprintf.ml#L515-L516

All that is missing to merge this PR is an official approval :)